producer.go

```diff
@@ -106,39 +113,80 @@ type AsyncApexLogNSQHandler struct {
 // be published to.
 //
 func NewAsyncApexLogNSQHandler(marshalFunc MarshalFunc, publishFunc PublishFunc, topic string, bufferSize int) *AsyncApexLogNSQHandler {
+	var backoff = 0
 	logChan := make(chan *log.Entry, bufferSize)
 	stopChan := make(chan bool, 1)
+
+	incrementBackoff := func(backoff int) int {
+		if backoff == 0 {
+			return 1
+		}
+		newBackoff := backoff * 2
+		if newBackoff > maximumBackoffMultiple {
+			return maximumBackoffMultiple
+		}
+		return newBackoff
+	}
+
+	handler := &AsyncApexLogNSQHandler{
+		logChan:  logChan,
+		stopChan: stopChan,
+	}
+
+	publishOrRetry := func(maxBackoff time.Duration, e *log.Entry) error {
```

Comment:
i think this flow of declaring a bunch of functions as local variables makes this a bit hard to read. the one of ideas behind the retryBackoff() example was to separate retry code from the business logic. if your backoff logic was contained in it's own function, then passed `publish` as an anonymous function into that, i think this would be a lot clearer.

Reply:
OK.. try now, it's a bit tidier.